lambda/matching.ml

```diff
@@ -2849,13 +2881,26 @@ let combine_constructor loc arg pat_env cstr partial ctx def
                 in
                 match act0 with
                 | Some act ->
+                    (* This case deviates from our policy, by typically
+                       generating three bytecode instructions.
+
+                       It can save a lot of bytecode space when matching
+                       on a type with many non-constant constructors,
+                       all sent to the same action. This pattern occurs
+                       several times in the compiler codebase
+                       (for example), due to code fragments such as the
+                       following:
+
+                           match token with SEMISEMI -> true | _ -> false
```

Comment:
That example is misleading.
IIUC, we only end up in this branch when all the non constant constructors are explicitly matched.

Reply:
I think that I have observed by testing that this example indeed reaches this branch, but it was some time ago so I need to re-test to make sure.

Reply:
I can confirm that I ran a test again, and for example the following case hits this part of the code:

```ocaml
let maybe_skip_phrase lexbuf =
  match !last_token with
  | Parser.SEMISEMI | Parser.EOF -> ()
  | _ -> skip_phrase lexbuf
```

My monitoring logic:

```ocaml
                    if nconstrs > 20 then
                      Printf.eprintf "DEBUG(%d) %s\n%!" nconstrs
                        (Debuginfo.Scoped_location.string_of_scoped_location loc);
```

Some hits:

DEBUG(123) Parse.skip_phrase
DEBUG(123) Parse.maybe_skip_phrase
DEBUG(110) Bytegen.preserve_tailcall_for_prim
DEBUG(68) Bytegen.weaken_event
DEBUG(123) Makedepend.lexical_approximation.skip_one